app/code/core/Mage/Core/Model/Resource.php

```diff
@@ -201,10 +203,14 @@ protected function _newConnection($type, $config)
      */
     protected function _getDefaultConnection($requiredConnectionName)
     {
+        $isAdmin = Mage::app()->getStore()->isAdmin();
+        $readResource = $isAdmin ? self::DEFAULT_ADMIN_READ_RESOURCE : self::DEFAULT_READ_RESOURCE;
+        $writeResource = $isAdmin ? self::DEFAULT_ADMIN_WRITE_RESOURCE : self::DEFAULT_WRITE_RESOURCE;
```

Comment:
This way the same function can for the same input return a different connection. If you send an email, the store is changed via emulation, probably not beeing Admin anymore? which means that existing transactions are ignored (or changed data from them not available) and, simply said a lot of hard to debug problems